### PR TITLE
Version 200

### PR DESCRIPTION
**Pull request by vinniefalco**
* Don't include OpenSSL for core snippets
* Tidy up msvc-14 workaround in multi_buffer
* buffers_cat fixes and coverage
* Refactor buffers_adaptor
* Refactor buffers_range
* Fix and refactor buffers_cat
* Refactor buffers_prefix
* Add const and mutable buffer sequence traits
* Add buffers_iterator_type trait
* Use new buffer traits, remove old unused traits
* Optimize for size on buffers_cat preconditions
* Refactor buffers_suffix
* Tidy up flat_buffer tests
* Fix ostream prepare calculation for low limits
* Tidy up flat_static_buffer tests
* Add more tests for dynamic buffers
* Tidy up multi_buffer
* Refactor ostream
* Refactor static_buffer

API Changes:

* buffers_adaptor replaces buffers_adapter (rename)
* make_printable replaces buffers (rename)
* Remove file_mode::append_new

Actions Required:

* Replace buffers_adapter.hpp with buffers_adaptor.hpp, and
  replace buffers_adapter with buffers_adaptor. Or, define
  BOOST_BEAST_ALLOW_DEPRECATED

* Replace call sites to use make_printable instead of buffers,
  and also include make_printable.hpp instead of ostream.hpp.

* Replace file_mode::append_new with file_mode::append
  or file_mode::append_existing instead of file_mode::append_new

### CODE REVIEW

File: include/boost/beast/core/buffer_traits.hpp [hunk 1]
```diff
@@ -12,24 +12,62 @@
...
 #include <type_traits>
 
 namespace boost {
 namespace beast {
 
-/** Type alias used to obtain the underlying buffer type of a buffer sequence.
+/** Determine if a list of types satisfy the <em>ConstBufferSequence</em> requirements.
+
+    This metafunction is used to determine if all of the specified types
+    meet the requirements for const buffer sequences. This type alias
```
Comment (djarek) [inline include/boost/beast/core/buffer_traits.hpp]: The documentation refers to "this type alias" whereas the user will see `struct ...`

File: include/boost/beast/core/buffer_traits.hpp [hunk 2]
```diff
@@ -12,24 +12,62 @@
...
+#else
+template<class... TN>
+using is_const_buffer_sequence = mp11::mp_all<
+    net::is_const_buffer_sequence<TN>...>;
+#endif
+
+/** Determine if a list of types satisfy the <em>MutableBufferSequence</em> requirements.
+
+    This metafunction is used to determine if all of the specified types
+    meet the requirements for mutable buffer sequences. This type alias
```
Comment (djarek) [inline include/boost/beast/core/buffer_traits.hpp]: The documentation refers to "this type alias" whereas the user will see `struct ...`

File: include/boost/beast/core/buffer_traits.hpp [hunk 3]
```diff
@@ -12,24 +12,62 @@
...
+template<class... TN>
+struct is_const_buffer_sequence : __see_below__ {};
+//using is_const_buffer_sequence = __see_below__;
+#else
+template<class... TN>
+using is_const_buffer_sequence = mp11::mp_all<
+    net::is_const_buffer_sequence<TN>...>;
+#endif
+
+/** Determine if a list of types satisfy the <em>MutableBufferSequence</em> requirements.
```
Comment (djarek) [inline include/boost/beast/core/buffer_traits.hpp]: `a list of types satisfy` -> `all types in T... satisfy`

File: include/boost/beast/core/buffer_traits.hpp (hunk 1 above)
Comment (vinniefalco) [reply include/boost/beast/core/buffer_traits.hpp]: The docs are right. "struct..." is a workaround for the XSLT not supporting type aliases yet.

File: include/boost/beast/core/buffer_traits.hpp (hunk 3 above)
Comment (vinniefalco) [reply include/boost/beast/core/buffer_traits.hpp]: Yeah that's better!

**Pull request closed without merging**